bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/CMakeMainTab2.java

```diff
@@ -203,9 +203,7 @@ private void updateArgumentsWithDefaultFlashCommand(ILaunchConfiguration configu
 					espFlashCommand);
 			argumentsForSerialFlash = argumentsForSerialFlash.isEmpty() ? espFlashCommand : argumentsForSerialFlash;
 			initializeJtagComboFields(configuration);
-			argumentsForJtagFlash = savedArgumentsForJtagFlash.contentEquals(savedArgumentsForJtagFlash)
-					? argumentsForJtagFlash
-					: savedArgumentsForJtagFlash;
+			argumentsForJtagFlash = savedArgumentsForJtagFlash;
 			if (isFlashOverJtag) {
 				argumentField.setText(argumentsForJtagFlash);
 			} else {
```

Comment:
I think we could improve this method a bit

You're already returning from method if it's not jtag flashing after setting `argumentField.setText(argumentsForSerialFlash);`
https://github.com/espressif/idf-eclipse-plugin/blob/852c21f1248cdab5574d2a164683a5d348578be4/bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/CMakeMainTab2.java#L198

so no need to do it again here https://github.com/espressif/idf-eclipse-plugin/blob/852c21f1248cdab5574d2a164683a5d348578be4/bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/CMakeMainTab2.java#L212


We could remove this entire from code [here](https://github.com/espressif/idf-eclipse-plugin/blob/852c21f1248cdab5574d2a164683a5d348578be4/bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/CMakeMainTab2.java#L206-L213) and keep it one statement `argumentField.setText(argumentsForJtagFlash);`

Reply:
Hi @kolipakakondal, thanks for the review. Refactored code a little bit and improved saving in the last commit